config/condition_schedule_test.go

```diff
@@ -0,0 +1,183 @@
+package config
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestScheduleConditionConfig_Copy(t *testing.T) {
+	t.Parallel()
+
+	cases := []struct {
+		name string
+		a    *ScheduleConditionConfig
+	}{
+		{
+			"nil",
+			nil,
+		},
+		{
+			"empty",
+			&ScheduleConditionConfig{},
+		},
+		{
+			"fully_configured",
+			&ScheduleConditionConfig{
+				Cron: String("* * * * * * *"),
+			},
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			r := tc.a.Copy()
+			if tc.a == nil {
+				// returned nil interface has nil type, which is unequal to tc.a
+				assert.Nil(t, r)
+			} else {
+				assert.Equal(t, tc.a, r)
+			}
+		})
+	}
+}
+
+func TestScheduleConditionConfig_Merge(t *testing.T) {
+	t.Parallel()
+
+	cases := []struct {
+		name string
+		a    *ScheduleConditionConfig
+		b    *ScheduleConditionConfig
+		r    *ScheduleConditionConfig
+	}{
+		{
+			"nil_a",
+			nil,
+			&ScheduleConditionConfig{},
+			&ScheduleConditionConfig{},
+		},
+		{
+			"nil_b",
+			&ScheduleConditionConfig{},
+			nil,
+			&ScheduleConditionConfig{},
+		},
+		{
+			"nil_both",
+			nil,
+			nil,
+			nil,
+		},
+		{
+			"empty",
+			&ScheduleConditionConfig{},
+			&ScheduleConditionConfig{},
+			&ScheduleConditionConfig{},
+		},
+		{
+			"cron_overrides",
+			&ScheduleConditionConfig{Cron: String("same")},
+			&ScheduleConditionConfig{Cron: String("different")},
+			&ScheduleConditionConfig{Cron: String("different")},
+		},
+		{
+			"cron_empty_one",
+			&ScheduleConditionConfig{Cron: String("same")},
+			&ScheduleConditionConfig{},
+			&ScheduleConditionConfig{Cron: String("same")},
+		},
+		{
+			"cron_empty_two",
+			&ScheduleConditionConfig{},
+			&ScheduleConditionConfig{Cron: String("same")},
+			&ScheduleConditionConfig{Cron: String("same")},
+		},
+		{
+			"cron_empty_same",
+			&ScheduleConditionConfig{Cron: String("same")},
+			&ScheduleConditionConfig{Cron: String("same")},
+			&ScheduleConditionConfig{Cron: String("same")},
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			r := tc.a.Merge(tc.b)
+			if tc.r == nil {
+				// returned nil interface has nil type, which is unequal to tc.r
+				assert.Nil(t, r)
+			} else {
+				assert.Equal(t, tc.r, r)
+			}
+		})
+	}
+}
+
+func TestScheduleConditionConfig_Finalize(t *testing.T) {
+	t.Parallel()
+
+	cases := []struct {
+		name string
+		s    []string
+		i    *ScheduleConditionConfig
+		r    *ScheduleConditionConfig
+	}{
+		{
+			"empty",
+			[]string{},
+			&ScheduleConditionConfig{},
+			&ScheduleConditionConfig{
+				Cron: String(""),
+			},
+		},
+	}
```

Comment:
For completeness, I think we should also test the other conditions, namely:
```
		{
			"empty",
			[]string{},
			&ScheduleConditionConfig{},
			&ScheduleConditionConfig{
				Cron: String(""),
			},
		},		
                 {
			"nil config",
			[]string{},
			nil,
			nil,
		},		
                {
			"chron povided",
			[]string{},
			&ScheduleConditionConfig{
				Cron: String("* * * * *"),
			},
			&ScheduleConditionConfig{
				Cron: String("* * * * *"),
			},
		},
```

Reply:
Good idea, added!